libraries/AP_Compass/AP_Compass_MAG3110.cpp

```diff
@@ -195,15 +199,15 @@ bool AP_Compass_MAG3110::_read_sample()
 }
 
 
-#define MAG_SCALE 1.0/10000  // 1 Tesla full scale of +-10000
+#define MAG_SCALE (1.0/10000)  // 1 Tesla full scale of +-10000
```

Comment:
could you change to 1.0f to ensure that we are using float calculation

Reply:
dot in 1.0 guarantees that this is float, so 1.0f is excessive tautology, oiled oil.

Reply:
1.0 means double in c++

Reply:
yes, if there is no key -fsingle-precision-constant in command line. But as you wish